designs/metrics.md

```diff
@@ -0,0 +1,296 @@
+import "github.com/Azure/karpenter-provider-azure/pkg/metrics"
+
+metrics.ComponentMetric.Inc(ctx, "description", metrics.Label(value))
+```
+
+**After:**
+```go
+// No separate metrics import needed - metrics are in same package
+
+ComponentMetric.WithLabelValues(value).Inc()
+```
+
+## Design Principles
+
+### Metric Namespace Identification
+
+To distinguish Azure provider metrics from core Karpenter metrics, we use a **different namespace**:
+
+- **Core Karpenter metrics**: `karpenter_*` (from upstream Karpenter)
+- **Azure provider metrics**: `karpenter_azure_*` (Azure-specific implementations)
+
+This is configured in `pkg/metrics/constants.go`:
+
+```go
+const (
+    Namespace = "karpenter_aks"  // Identifies Azure provider metrics
+)
+```
+
+**Why This Matters:**
+- **Clear separation** between core and provider-specific metrics
+- **Avoids conflicts** with upstream Karpenter metric names
+- **Easy filtering** in monitoring systems (e.g., `karpenter_aks_*` for Azure-only metrics)
+- **Consistent naming** across all Azure provider components
+
+**Metric Naming Examples:**
+- `karpenter_aks_instance_vm_create_start_total` - Azure provider metric
+- `karpenter_nodeclaims_created_total` - Core Karpenter metric
+
+### Co-location Pattern
+
+Following Kubernetes best practices, **production Prometheus metrics are co-located with their implementation components**:
+
+1. **Provider-specific metrics** → Live within provider packages (e.g., `pkg/providers/instance/`)
+2. **Controller-specific metrics** → Live within controller packages (e.g., `pkg/controllers/*/`)
+3. **Component-specific metrics** → Live with the component (batcher, cache, etc.)
+
+### Why Co-location?
+
+#### 1. **Locality of Reference**
+- Metrics are defined next to the code that uses them
+- Easy to understand what metrics a component exposes
+- Reduces cognitive load when reading/maintaining code
+
+#### 2. **Package Encapsulation**
+- Each component owns its metrics
+- Changes to component behavior and metrics happen together
+- No cross-package dependencies for metrics
+
+## Specific Changes in Detail
+
+### Label Design
+
+Each component defines **labels relevant to its operations**:
+
+```go
+// Example from instance provider
+ComponentMetric = prometheus.NewCounterVec(
+    prometheus.CounterOpts{
+        Namespace: coremetrics.Namespace,  // From centralized constants
+        Subsystem: componentSubsystem,
+        Name:      "operation_total",
+        Help:      "Total number of operations.",
+    },
+    []string{"label1", "label2"},  // Component-specific labels
+)
+```
+
+**Label Selection Guidelines:**
+- Choose labels that help diagnose component-specific issues
+- Keep cardinality reasonable (avoid high-cardinality labels like instance IDs)
+- Use labels that align with how the component operates
+
+### Label Constants
+
+Defined locally in each component's metrics file:
+
+```go
+const (
+    componentSubsystem = "component_name"
+    label1 = "label_one"
```

Comment:
may be worth calling out explicit expectation of `snake_case` in labels, and reference to prom docs where they recommend that (I assume it exists somewhere?)